pkg/controller/controller.go

```diff
@@ -620,6 +620,19 @@ func (c *controller) updateInstanceFinalizers(
 	instance *v1alpha1.Instance,
 	finalizers []string) error {
 
+	// Get the latest version of the instance so that we can avoid conflicts
+	// (since we have probably just updated the status of the instance and are
+	// now removing the last finalizer).
+	//
+	// It's important to do everything we can to ensure that we complete this
```

Comment:
If you feel that is more information than necessary, i'm happy to just take it out.  TLDR of the comment is, we just updated this thing so we need to get the current state to avoid conflicts.